android/app/src/main/java/org/fossasia/openevent/viewmodels/SpeakersListFragmentViewModel.java

```diff
@@ -19,17 +19,22 @@
     private RealmDataRepository realmRepo;
     private RealmResults<Speaker> realmResults;
     private String searchText = "";
+    private int speakersListSortType = 0;
 
     public SpeakersListFragmentViewModel() {
         realmRepo = RealmDataRepository.getDefaultInstance();
     }
 
-    public LiveData<List<Speaker>> getSpeakers() {
-        speakersList = new MutableLiveData<>();
-        realmResults = realmRepo.getSpeakers(sortOrderSpeaker());
-        realmResults.addChangeListener((speakers, orderedCollectionChangeSet) -> {
-            speakersList.setValue(speakers);
-        });
+    public LiveData<List<Speaker>> getSpeakers(int sortType) {
+        if (speakersList == null || sortType != speakersListSortType) {
+            speakersList = new MutableLiveData<>();
```

Comment:
new List should only be created if it is null. Make data holders as immutable as possible

Reply:
The issue is with the implementation of the Realm Repository. It sends the list of speakers according to the sort order passed to it. If the sort order is changed , then the LiveData containing the previous list becomes useless.

Reply:
@iamareebjamal An alternative would be that we can store all the three type of List as LiveData in the ViewModel return the appropriate one according to the sort order. But this would mean more memory consumption.

Reply:
You should clear instead of creating new